lib/modules/tests/index.js

```diff
@@ -0,0 +1,239 @@
+              const _next = () => { next(null, results); };
+              if (options.noBrowser) {
+                return next(null, results);
+              }
+              opn(fs.dappPath('coverage/__root__/index.html'), {wait: false})
+                .then(() => new Promise(resolve => setTimeout(resolve, 1000)))
+                .then(_next, _next);
+            });
+        });
+      }
+    ], (err, results) => {
+      if (err) {
+        self.logger.error(err);
+        return cb(err);
+      }
+      let totalFailures = results.reduce((acc, result) => acc + result.failures, 0);
+      if (totalFailures) {
+        return cb(` > Total number of failures: ${totalFailures}`.red.bold);
+      }
+      console.info(' > All tests passed'.green.bold);
+      cb();
+    });
+  }
+
+
+  getFilesFromDir(filePath, cb) {
+    const self = this;
+
+    fs.stat(filePath, (err, fileStat) => {
+      const errorMessage = `File "${filePath}" doesn't exist or you don't have permission to it`.red;
+      if (err) {
+        return cb(errorMessage);
+      }
+      let isDirectory = fileStat.isDirectory();
+      if (isDirectory) {
+        return fs.readdir(filePath, (err, files) => {
+          if (err) {
+            return cb(err);
+          }
+          async.map(files, (file, _cb) => {
+            self.getFilesFromDir(path.join(filePath, file), _cb);
+          }, (err, arr) => {
+            if (err) {
+              return cb(errorMessage);
+            }
+            cb(null, arr.reduce((a,b) => a.concat(b), []));
+          });
+        });
+      }
+      cb(null, [filePath]);
+    });
+  }
+
+  runJSTests(files, options, cb) {
+    const self = this;
+    const loglevel = options.loglevel || 'warn';
+    async.waterfall([
+      function setupGlobalNamespace(next) {
+        const test = new Test({loglevel, node: options.node, events: self.events, logger: self.logger,
+          config: self.embark.config, ipc: self.ipc});
+        global.embark = test;
+        global.assert = assert;
+        global.config = test.config.bind(test);
+
+        let deprecatedWarning = function () {
+          self.logger.error(__('%s are not supported anymore', 'EmbarkSpec & deployAll').red);
+          self.logger.error(__('You can learn about the new revamped tests here: %s', 'https://embark.status.im/docs/testing.html'.underline));
+          process.exit();
+        };
+
+        global.deployAll = deprecatedWarning;
+        global.EmbarkSpec = {};
+        global.EmbarkSpec.deployAll = deprecatedWarning;
+
+        // Override require to enable `require('Embark/contracts/contractName');`
+        const Module = require('module');
+        const originalRequire = require('module').prototype.require;
+        Module.prototype.require = function (requireName) {
+          if (requireName.startsWith('Embark/contracts')) {
+            return test.require(...arguments);
+          }
+          return originalRequire.apply(this, arguments);
+        };
```

Comment:
😱

Reply:
Welcome to the world of Magic :sparkles: